src/features/user/infrastructure.rs

```diff
@@ -1,15 +1,18 @@
-use crate::app::AppContext;
+use crate::{app::AppContext, features::user::errors::UserError};
 use sqlx::Row;
 
 use super::entities::{User, Vote};
 
-pub(crate) async fn create_user_in_db(
-    app_ctx: &AppContext,
-    user: User,
-) -> Result<User, sqlx::Error> {
-    let mut pool = app_ctx.infrastructure().get_repository().await?;
+pub(crate) async fn create_user_in_db(app_ctx: &AppContext, user: User) -> Result<User, UserError> {
+    let mut pool = match app_ctx.infrastructure().get_repository().await {
+        Ok(p) => p,
+        Err(error) => {
+            tracing::error!("{error:?}");
```

Comment:
In some spots we've got `tracing:errror!(...)` and in others simply `error!(...)`. Happy for either style, but lets be consistent please 🙏